apps/mull-ui/src/app/pages/create-event/create-event.tsx

```diff
@@ -0,0 +1,238 @@
+import { useFormik } from 'formik';
+import { toast, TypeOptions } from 'react-toastify';
+import * as Yup from 'yup';
+
+import { EventRestriction } from '@mull/types';
+import { PillOptions, CustomTextInput, CustomTimePicker } from '@mull/ui-lib';
+import DateCalendar from '../create-event/date-calendar/date-calendar';
+
+import { DAY_IN_MILLISECONDS } from '../../../constants';
+
+import { ReactComponent as PencilIcon } from '../../../assets/icons/event-form-icons/pencil.svg';
+import { ReactComponent as DescriptionIcon } from '../../../assets/icons/event-form-icons/description.svg';
+import { ReactComponent as LocationIcon } from '../../../assets/icons/event-form-icons/pin.svg';
+import { ReactComponent as UploadIcon } from '../../../assets/icons/event-form-icons/upload.svg';
+
+import './create-event.scss';
+
+export interface CreateEventProps {
+  history: History;
+}
+
+// Mutation to create events
+const CREATE_EVENT = gql`
+  mutation CreateEvent($createEventInput: CreateEventInput!) {
+    createEvent(createEventInput: $createEventInput) {
+      id
+    }
+  }
+`;
+
+/**
+ * This component renders the create event page
+ * @param {History} history
+ */
+const CreateEventPage = ({ history }) => {
+  // Reference Id for the toast
+  const toastId = useRef(null);
+  // GraphQL mutation hook to create events
+  const [createEvent] = useMutation(CREATE_EVENT);
+  // Image of Event
+  const [imageFile, setImageFile] = useState(null);
+
+  /**
+   * Handles image file uploads
+   * @param {ChangeEvent<HTMLInputElement>} event
+   */
+  const handleFileUpload = (event: ChangeEvent<HTMLInputElement>) => {
+    setImageFile(URL.createObjectURL(event.target.files[0]));
+  };
+
+  /**
+   * Adds time to a Date object
+   * @param {string} time Time in HH:MM format
+   * @param {Date} date Date object
+   */
+  const addTimeToDate = (time: string, date: Date) => {
+    const [hour, minute] = time.split(':');
+    date.setHours(parseInt(hour));
+    date.setMinutes(parseInt(minute));
+  };
+
+  /**
+   * Notifies the user of a event submission
+   */
+  const notifySubmissionToast = () => {
+    toastId.current = toast('Submitting Event...', { autoClose: false });
+  };
+  /**
+   * Updates existing toast.
+   * @param {TypeOptions} type Type of the toast
+   * @param {string} message Message to display
+   */
+  const updateSubmissionToast = (type: TypeOptions, message: string) => {
+    toast.update(toastId.current, {
+      type,
+      render: message,
+      autoClose: 3000,
+    });
+  };
+
+  const formik = useFormik({
+    initialValues: {
+      activeRestriction: EventRestriction.NONE,
+      startDate: null,
+      endDate: null,
+      startTime: '',
+      endTime: '',
+      eventTitle: '',
+      description: '',
+      location: '',
+    },
+
+    validationSchema: Yup.object({
+      startDate: Yup.date().required('Start date is required'),
+      endDate: Yup.date()
+        .nullable()
+        .required('End date is required')
```

Comment:
I don't think that having a null endDate should be an error. If the user didn't set the endDate, I think its safe to assume that they want a one-day event.

Reply:
Agreed, choosing a single date wasn't intuitive because I had to tap the same date twice.

Reply:
I left it as nullable and will set it as the start date in the submit logic.